test/e2e/scenario/recorder.scenario.ts

```diff
@@ -654,6 +670,236 @@ describe('recorder', () => {
         expect(segment.records.slice(3).every((record) => record.type !== RecordType.FullSnapshot)).toBe(true)
       })
   })
+
+  describe('layout viewport properties - ', () => {
+    createTest('getWindowWidth/Height reports using layout viewport dimensions')
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        /**
+         * The purpose of this test is to check that browser dimensions are measuring the
+         * layout viewport (not visual viewport), and so they do not change when pinch zoom is applied.
+         *
+         * InnerWidth/Height on some devices/browsers are changed by pinch zoom.
+         * Read vendor discussions around the challenges of standardization:
+         * - https://bugs.chromium.org/p/chromium/issues/detail?id=489206
+         */
+        if (isGestureUnsupported()) {
+          return // No Fallback test
+        }
+        await resetViewport()
+
+        const { innerWidth, innerHeight } = (await browserExecute(() => ({
+          innerWidth: window.innerWidth,
+          innerHeight: window.innerHeight,
+        }))) as { innerWidth: number; innerHeight: number }
+
+        const initialVisualViewport = await getVisualViewport()
+        await pinchZoom(150)
+        await pinchZoom(150)
+        await browser.pause(210)
+        const nextVisualViewport = await getVisualViewport()
+
+        await browserExecute(() => {
+          window.dispatchEvent(new Event('resize'))
+        })
+        await browser.pause(210)
+
+        await flushEvents()
+        const segment = getLastSegment(events)
+        const ViewportResizeRecords = findAllIncrementalSnapshots(segment, IncrementalSource.ViewportResize)
+        const lastViewportResizeRecord = ViewportResizeRecords.slice(-1)[0].data as ViewportResizeData
+
+        // Mac OS X Chrome scrollbars are included here (~15px) which seems to be against spec
+        // Scrollbar edge-case handling not considered right now, further investigation needed
+        let scrollbarWidth = 0
+        if (browser.capabilities.browserName === 'chrome' && browser.capabilities.platformName === 'mac os x') {
+          scrollbarWidth = await getScrollbarWidth()
+        }
+        expectToBeNearby(lastViewportResizeRecord.width, innerWidth - scrollbarWidth)
+        expectToBeNearby(lastViewportResizeRecord.height, innerHeight - scrollbarWidth)
+
+        // Test the test: ensure the pinch zoom worked
+        expect(initialVisualViewport.scale < nextVisualViewport.scale).toBeTruthy()
```

Comment:
it would be way easier to read your test by abstracting some operations.
FMU, we could reduce your test to something like:
```suggestion
        const layoutViewport = await getLayoutViewport()
        await performSignificantZoom()
        const resizeRecord = await sendResizeRecord(events)
        const scrollbarCorrection = await getScrollbarCorrection()

        expectToBeNearby(resizeRecord.width, layoutViewport.width - scrollbarCorrection)
        expectToBeNearby(resizeRecord.height, layoutViewport.height - scrollbarCorrection)
```
wdyt?